reconcile/utils/gql.py

```diff
@@ -198,13 +193,39 @@ def get_resource(path: str) -> dict[str, Any]:
     return get_api().get_resource(path)
 
 
+class PersistentRequestsHTTPTransport(RequestsHTTPTransport):
+    def connect(self):
```

Comment:
Is there a way to reconnect in case of connection being deleted due to TCP idle time out?

Reply:
I will test this locally. But I think the name of the method chosen here is irritating since HTTP is stateless...  Also, the GQL lib is using requests.Session, which handles the connections.

Reply:
It would be great if you could test it locally, e.g. sleeping between the first two queries and making sure it reconnects. How much sleep time? The greater the better. Apparently [the default is 2 hours](https://man7.org/linux/man-pages/man7/tcp.7.html#:~:text=The%20default%20value%20is%207200,timeouts%20may%20be%20much%20shorter.), you may want to change that in your tests 😅

Reply:
I simply restart nginx proxy during test, looks good:

```
[2022-05-19 14:03:56] [DEBUG] [openshift_resources_base.py:fetch_openshift_resource:478] - Fetching resource: /observability/prometheusrules/mt-sre-ci-int.prometheusrules.yaml
[2022-05-19 14:03:56] [DEBUG] [connectionpool.py:_get_conn:271] - Resetting dropped connection: localhost
[2022-05-19 14:03:57] [DEBUG] [connectionpool.py:_new_conn:939] - Starting new HTTPS connection (2): localhost:40444
```